.github/workflows/deployment.yml

```diff
@@ -164,21 +164,57 @@ jobs:
           --java-options --add-opens=javafx.base/javafx.collections.transformation=org.jabref \
           --java-options --add-modules=jdk.incubator.vector
 
-          # Add additional files / directories
-          cp buildres/mac/jabrefHost.py build/distribution/JabRef.app/Contents/
-          cp -R buildres/mac/native-messaging-host build/distribution/JabRef.app/Contents/
-
-          # Sign the final artifact
-          codesign --force --deep --sign "Developer ID Application: JabRef e.V. (6792V39SK3)" \
-            --entitlements buildres/mac/jabref.entitlements \
-            --options runtime --timestamp build/distribution/JabRef.app
+      - name: Build pkg (macOS)
+        if: ((matrix.os == 'macos-13') || (matrix.os == 'macos-14')) && (steps.checksecrets.outputs.secretspresent == 'YES')
+        shell: bash
+        run: |
+          set -e
+          cd jabgui
 
-          # pack dmg and pkg
-          hdiutil create -volname "JabRef" -srcfolder build/distribution/JabRef.app -ov -format UDZO build/distribution/JabRef${{ matrix.suffix }}.dmg
-          productbuild --component build/distribution/JabRef.app /Applications --sign "Developer ID Installer: JabRef e.V. (6792V39SK3)" build/distribution/JabRef${{ matrix.suffix }}.pkg
+          jpackage \
+          --module org.jabref/org.jabref.Launcher \
+          --module-path ${{env.JAVA_HOME}}/jmods/:build/jlinkbase/jlinkjars \
+          --add-modules org.jabref,org.jabref.merged.module  \
+          --add-modules jdk.incubator.vector \
+          --dest build/distribution \
+          --app-content buildres/mac/Resources/jabrefHost.py \
+          --app-content buildres/mac/Resources/native-messaging-host \
```

Comment:
According to https://github.com/openjdk/jdk/pull/21698#issuecomment-2849491184 there may only be one: a directory